### PR TITLE
nat division by timedelta

### PR DESCRIPTION
**Pull request by jbrockmendel**
xref #17876 

Does the whatsnew note still go in 0.20.0.txt?

- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -830,6 +830,7 @@ Other API Changes
 - Restricted DateOffset keyword arguments.  Previously, ``DateOffset`` subclasses allowed arbitrary keyword arguments which could lead to unexpected behavior.  Now, only valid arguments will be accepted. (:issue:`17176`).
 - Pandas no longer registers matplotlib converters on import. The converters
   will be registered and used when the first plot is draw (:issue:`17710`)
+- ``NaT`` division with :class:`datetime.timedelta` will now return NaN instead of raising (:issue:`17876`)
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.21.0.txt]: Double-tick the NaN.

Comment (gfyoung) [reply doc/source/whatsnew/v0.21.0.txt]: Also, what is the rationale for returning `NaN` ?  Why not return `NaT` instead?

Comment (gfyoung) [reply doc/source/whatsnew/v0.21.0.txt]: It also occurs to me that you *might* not need a `whatsnew` given that this is internal.

Comment (jbrockmendel) [reply doc/source/whatsnew/v0.21.0.txt]: > Double-tick the NaN. 

Just pushed a fix.

> Also, what is the rationale for returning `NaN` ? Why not return `NaT` instead?

This is just taking behavior that currently applies to `Timedelta` and applying the same to `timedelta`.

Comment (gfyoung) [reply doc/source/whatsnew/v0.21.0.txt]: Hmm...seems a little non-intuitive to me, but then again, I'm not going to push changing the norm without getting feedback from other maintainers:

@jreback @jorisvandenbossche ?

Comment (jbrockmendel) [reply doc/source/whatsnew/v0.21.0.txt]: I think the idea is that in this context `NaT` is ducking as a `Timedelta`, so `Y = NaT / Timedelta(X)` is solving for `Timedelta(X) * Y = NaT` --> `Y = np.nan`

Comment (gfyoung) [reply doc/source/whatsnew/v0.21.0.txt]: Ah, I see.  Right, that makes sense.

Comment (jorisvandenbossche) [reply doc/source/whatsnew/v0.21.0.txt]: Yes, that seems to make sense to me, similar as:

```
In [5]: np.nan / 5
Out[5]: nan
```

And also this is how `pd.NaT / pd.Timedelta(..)` is already working? In that case I would rather call it a bug fix.

Comment (jbrockmendel) [reply doc/source/whatsnew/v0.21.0.txt]: > And also this is how pd.NaT / pd.Timedelta(..) is already working? 

Yes.

**Comment by jreback** [comment]
will be for 0.21.1, we haven't pushed the whatsnew yet, so will need to update then.

**Comment by jbrockmendel** [comment]
> will be for 0.21.1, we haven't pushed the whatsnew yet, so will need to update then.

OK.  Does that mean I need to something now or just be patient?

**Comment by gfyoung** [comment]
@jbrockmendel : Just hold on for the time being.  This PR is good to go otherwise I think.  We'll ping you once it's added, or if we don't, ping us about it.

File: doc/source/whatsnew/v0.21.0.txt [hunk 2]
```diff
@@ -830,6 +830,7 @@ Other API Changes
 - Restricted DateOffset keyword arguments.  Previously, ``DateOffset`` subclasses allowed arbitrary keyword arguments which could lead to unexpected behavior.  Now, only valid arguments will be accepted. (:issue:`17176`).
 - Pandas no longer registers matplotlib converters on import. The converters
   will be registered and used when the first plot is draw (:issue:`17710`)
+- ``NaT`` division with :class:`datetime.timedelta` will now return ``NaN`` instead of raising (:issue:`17876`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: so will move to 0.21.1 when its available

Comment (jreback) [reply doc/source/whatsnew/v0.21.0.txt]: can you move to 0.22.0 other api changes

Comment (jbrockmendel) [reply doc/source/whatsnew/v0.21.0.txt]: Done.  Out of curiosity, why 0.22.0 instead of 0.21.1?

**Comment by jreback** [comment]
@jbrockmendel if you would rebase

**Comment by jreback** [comment]
ok ping on green.

**Comment by jreback** [comment]
thanks

**Pull request merged**